modvendor/modvendor_report.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package modvendor
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/tf"
+)
+
+// Vendored describes a vendored dependency.
+type Vendored struct {
+	// Source is the remote source.
+	Source tf.Source
+	// Dir is the directory where the dependency have been vendored into.
+	Dir string
+}
+
+// IgnoredVendor describes an ignored dependency.
+type IgnoredVendor struct {
+	RawSource string
+	Reason    string
+}
+
+// Report with the result of the vendor related functions.
+type Report struct {
+	Vendored map[string]Vendored
+	Ignored  []IgnoredVendor
+	Error    error
+}
+
+// NewEmptyReport returns a new empty report.
+func NewEmptyReport() Report {
```

Comment:
Maybe just NewReport ? it is kinda usual that New functions create an empty new instance.

Reply:
the original idea was using `NewReport(vendored, ignored, err)` but in the end it wasn't needed, so yeah, I will apply your suggestion. Thanks.